lib/matplotlib/patches.py

```diff
@@ -4589,23 +4589,27 @@ def _get_xy(self, xy, s, axes=None):
         s0 = s  # For the error message, if needed.
         if axes is None:
             axes = self.axes
-        xy = np.array(xy)
+
+        # preserve mixed type input (such as str, int)
+        x = np.array(xy[0])
+        y = np.array(xy[1])
+
         fig = self.get_figure(root=False)
         if s in ["figure points", "axes points"]:
-            xy *= fig.dpi / 72
+            x *= fig.dpi / 72
+            y *= fig.dpi / 72
             s = s.replace("points", "pixels")
         elif s == "figure fraction":
             s = fig.transFigure
         elif s == "subfigure fraction":
             s = fig.transSubfigure
         elif s == "axes fraction":
             s = axes.transAxes
-        x, y = xy
 
         if s == 'data':
             trans = axes.transData
-            x = float(self.convert_xunits(x))
-            y = float(self.convert_yunits(y))
+            x = cbook._to_unmasked_float_array(self.convert_xunits(x, axes))
+            y = cbook._to_unmasked_float_array(self.convert_yunits(y, axes))
```

Comment:
If we want to use an explicit `Axes`, do we need to go through `Artist.convert_xunits` (with the addition of the `axes` parameter), or instead call `Axes.xaxis.convert_units` ourselves with the correct `axes`?